src/Traits/Connector/SendsRequests.php

```diff
@@ -24,20 +21,87 @@ trait SendsRequests
     /**
      * Send a request
      *
+     * @param callable(\Throwable, \Saloon\Contracts\Request): (bool)|null $handleRetry
      * @throws \ReflectionException
      * @throws \Throwable
      */
-    public function send(Request $request, MockClient $mockClient = null): Response
+    public function send(Request $request, MockClient $mockClient = null, callable $handleRetry = null): Response
     {
-        $pendingRequest = $this->createPendingRequest($request, $mockClient);
+        $maxTries = $request->tries ?? $this->tries ?? 1;
+        $retryInterval = $request->retryInterval ?? $this->retryInterval ?? 0;
```

Comment:
Similar to how you've defaulted `$maxTries` to be `1` on line 35 if the value is invalid, do you think it's worth doing the same with the `$retryInterval`? You could maybe default it to `0` if negative numbers are passed in?

From having a quick tinker, it looks like `usleep` (used on line 51) will throw an exception if you pass a negative number to it. Maybe that's the desired behaviour that you want?

But I just wondered whether you'd want to keep it consistent so that if either the `$maxTries` or `$retryInterval` are invalid, they'll be both to reset to default values? Or maybe they should both throw exceptions to highlight to the dev that they've not set something up right in their connectors or requests? At the moment, I think those two fields handle invalid values differently from each other.

This might just be a bit nitpicky though, so feel free to disregard it! 😄

Reply:
This is a good point! Yep I'll add a minimum in here as well. You can't sleep from a negative number!